src/js/player.js

```diff
@@ -3031,6 +3031,42 @@ class Player extends Component {
    *         A promise with a Picture-in-Picture window.
    */
   requestPictureInPicture() {
+    if (this.options_.preferDocumentPictureInPicture && 'documentPictureInPicture' in window) {
+      const pipContainer = document.createElement(this.el().tagName);
+
+      pipContainer.classList = this.el().classList;
+      pipContainer.classList.add('vjs-pip-container');
+      if (this.posterImage) {
+        pipContainer.appendChild(this.posterImage.el().cloneNode(true));
+      }
+      if (this.titleBar) {
+        pipContainer.appendChild(this.titleBar.el().cloneNode(true));
+      }
+      pipContainer.appendChild(Dom.createEl('p', { className: 'vjs-pip-text' }, {}, this.localize('Playing in picture-in-picture')));
+
+      window.documentPictureInPicture.requestWindow({
+        // This aspect ratio doesn't seem to be respected
```

Comment:
Do you see this on macOS?

Reply:
Yes at https://deploy-preview-8113--videojs-preview.netlify.app/sandbox/docpip.html on Canary 112

In the original page `videoWidth / videoHeight` is 2.4. in the PiP window `innerWidth / innerHeight` is 2.6666666666666665.

A 16/9 video's window is also too wide. I suppsose the aspect ratio is for the outer size of the window, including its title bar?

Reply:
I think you're hitting https://bugs.chromium.org/p/chromium/issues/detail?id=1407629 then. I hope it will be fixed soon.

FYI @liberato-at-chromium

Reply:
The PiP window at https://document-picture-in-picture-api.glitch.me/ is also too wide. It would be more useful to be able to set the inner size / aspect ratio. The title bar height will differ by OS, font settings, locale, and eventually browser.

Reply:
That is good feedback! I'll add it to the Chromium bug.

Reply:
Thanks. I was going to add an issue at WICG/document-picture-in-picture but won't then if it's tracked elsewhere.

Reply:
Filing an issue to the spec would help as well actually. Feel free to do so.

Reply:
Thanks for filing https://github.com/WICG/document-picture-in-picture/issues/25